lib/rubocop/cop/style/mixin_usage.rb

```diff
@@ -61,6 +61,8 @@ def on_send(node)
         private
 
         def top_level_node?(node)
+          return if node.class_type?
+
           if node.parent.parent.nil?
             node.sibling_index.zero?
```

Comment:
I guess using `sibling_index` is not correct to check top level.
For example, this cop does not add offenses to the code.


```ruby
def foo
end
include A # This include's `.parent.parent` is a nil, and the sibling index is not zero.
```

```bash
$ rubocop --only Style/MixinUsage
Inspecting 1 file
.

1 file inspected, no offenses detected
```


Maybe we can use `SendNode#macro?` instead. 

https://github.com/bbatsov/rubocop/blob/8459e201400f852df7c8ac01dd2ad5e760c092d5/lib/rubocop/ast/node/mixin/method_dispatch_node.rb#L32-L41

Reply:
It was this method that I really wanted🙏  This also can remove some unnecessary conditions.